frontend/express/public/javascripts/countly/countly.vue.components.js

```diff
@@ -372,4 +373,2886 @@
+                    });
+
+                    $.datepicker.setDefaults($.datepicker.regional[""]);
+                    self.dateTo.datepicker("option", $.datepicker.regional[countlyCommon.BROWSER_LANG]);
+                    self.dateFrom.datepicker("option", $.datepicker.regional[countlyCommon.BROWSER_LANG]);
+                },
+                toggle: function() {
+                    this.isOpened = !this.isOpened;
+                },
+                cancel: function() {
+                    this.isOpened = false;
+                },
+                setCustomPeriod: function() {
+                    var self = this;
+                    if (!self.dateFromSelected && !self.dateToSelected) {
+                        return false;
+                    }
+                    this.setPeriod([
+                        self.dateFromSelected - countlyCommon.getOffsetCorrectionForTimestamp(self.dateFromSelected),
+                        self.dateToSelected - countlyCommon.getOffsetCorrectionForTimestamp(self.dateToSelected) + 24 * 60 * 60 * 1000 - 1
+                    ]);
+                },
+                setPeriod: function(newPeriod) {
+                    countlyCommon.setPeriod(newPeriod);
+                    this.$root.$emit("cly-date-change");
+                    this.isOpened = false;
+                },
+                dateFromInputSubmit: function() {
+                    var date = moment(this.dateFromLabel, "MM/DD/YYYY");
+                    if (date.format("MM/DD/YYYY") === this.dateFromLabel) {
+                        this.dateFromSelected = date.valueOf();
+                    }
+                    else {
+                        this.dateFromLabel = moment(this.dateFromSelected).format("MM/DD/YYYY");
+                    }
+                },
+                dateToInputSubmit: function() {
+                    var date = moment(this.dateToLabel, "MM/DD/YYYY");
+                    if (date.format("MM/DD/YYYY") === this.dateToLabel) {
+                        this.dateToSelected = date.valueOf();
+                    }
+                    else {
+                        this.dateToLabel = moment(this.dateToSelected).format("MM/DD/YYYY");
+                    }
+                }
+            },
+            template: '<div class="cly-vue-global-date-selector-w help-zone-vs">\
+                            <div class="calendar inst-date-picker-button" @click="toggle" v-bind:class="{active: isOpened}" >\
+                                <i class="material-icons">date_range</i>\
+                                <span class="inst-selected-date">{{currentPeriodLabel}}</span>\
+                                <span class="down ion-chevron-down"></span>\
+                                <span class="up ion-chevron-up"></span>\
+                            </div>\
+                            <div class="inst-date-picker" v-show="isOpened">\
+                                <div class="date-selector-buttons">\
```

Comment:
Codacy found an issue: [Bad escaping of EOL. Use option multistr if needed.](https://app.codacy.com/manual/ar2rsawseen/countly-server/pullRequest?prid=6339769)